### PR TITLE
mm/iob: new function iob_reserve() to adjust headroom offset

### PR DESCRIPTION
**Pull request by anchao**
## Summary

mm/iob: new function iob_reserve() to adjust headroom offset
mm/iob: new function iob_update_pktlen() to update packet length of the iob

## Impact

N/A

## Testing

CI-check

### CODE REVIEW

File: mm/iob/iob_reserve.c [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+/****************************************************************************
+ * Name: iob_reserve
+ *
+ * Description:
+ *   Adjust headroom offset of iobs by reducing the tail room.
+ *
+ ****************************************************************************/
+
+FAR void iob_reserve(FAR struct iob_s *iob, unsigned int reserved)
```
[the remark this answers is not in the document]
Comment (anchao) [reply mm/iob/iob_reserve.c]: Done

File: mm/iob/iob_update_pktlen.c [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+        {
+          len = pktlen;
+        }
+
+      next->io_len = len;
+      pktlen      -= len;
+      next         = next->io_flink;
+    }
+
+  if (pktlen > 0)
```
Comment (xiaoxiang781216) [inline mm/iob/iob_update_pktlen.c:129]: impossible?

File: mm/iob/iob_reserve.c [hunk 3]
```diff
@@ -39,7 +39,7 @@
  *
  ****************************************************************************/
 
-FAR void iob_reserve(FAR struct iob_s *iob, unsigned int reserved)
```
Comment (xiaoxiang781216) [inline mm/iob/iob_reserve.c]: move to first patch

File: mm/iob/iob_update_pktlen.c (hunk 2 above)
Comment (anchao) [reply mm/iob/iob_update_pktlen.c]: Done

File: mm/iob/iob_reserve.c (hunk 3 above)
Comment (anchao) [reply mm/iob/iob_reserve.c]: Done

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**